### PR TITLE
search: fix pagination

### PR DESCRIPTION
**Pull request by jiri-fiala**
* Fixes start index field name mismatch
* Deduplicates the `hcsearch` fuction
* Avoids absolute search javascript referencing

Signed-off-by: Jiri Fiala <jfiala@redhat.com>

### CODE REVIEW

**Comment by kalexand-rh** [comment]
@vikram-redhat?

File: search-commercial.html [hunk 1]
```diff
@@ -186,6 +186,7 @@ <h1>Search Results</h1>
     </div>
 </footer>
 
+<script src="container-platform/4.2/_javascripts/hc-search.js" type="text/javascript"></script>
```
Comment (vikram-redhat) [inline search-commercial.html]: @jiri-fiala missing a /

File: _templates/_page_openshift.html.erb [hunk 2]
```diff
@@ -127,10 +127,10 @@
   <script src="https://assets.openshift.net/content/modernizr.js" type="text/javascript"></script>
   <script src="https://assets.openshift.net/content/subdomain.js" type="text/javascript"></script>
   <script src="https://assets.openshift.net/content/nav-tertiary.js" type="text/javascript"></script>
-  <script src="https://docs.openshift.com/container-platform/4.1/_javascripts/bootstrap-offcanvas.js" type="text/javascript"></script>
-  <script src="https://docs.openshift.com/container-platform/4.1/_javascripts/reformat-html.js" type="text/javascript"></script>
-  <script src="https://docs.openshift.com/container-platform/4.1/_javascripts/hc-search.js" type="text/javascript"></script>
-  <script src="https://docs.openshift.com/container-platform/4.1/_javascripts/page-loader.js" type="text/javascript"></script>
+  <script src="/container-platform/4.2/_javascripts/bootstrap-offcanvas.js" type="text/javascript"></script>
```
Comment (vikram-redhat) [inline _templates/_page_openshift.html.erb]: This won't work for docs.okd.io or (I think) ARO docs. Further I prefer 4.1 as that is a reference point (first release for 4.x). I understand that not doing this makes it harder to test locally but hard coding the location of the scripts was the only way I could make it work at 4.1 release time with changes to our asciibinder script. Sorry! :(

Comment (jiri-fiala) [reply _templates/_page_openshift.html.erb]: Fair point and thanks for catching this. I have reverted changes to this file, keeping the original hard-coded locations of the scripts. I have also used the same style (absolute 4.1 path) in order to source the script in `search-commercial.html` for consistency.

**Comment by jiri-fiala** [comment]
@vikram-redhat, I'm not sure, but this is maybe waiting for your re-review after I addressed the comments?

**Pull request merged**